### PR TITLE
clang-format: add a whitelist

### PR DESCRIPTION
**Pull request by nerzhul**
If file is in the whitelist, softfail, else hard failure

Some files are not in whitelist and marked as normal:
* src/content_mapnode.h
* src/cguittfont/xCGUITTFont.cpp
* src/gameparams.h
* src/profiler.cpp

### CODE REVIEW

**Comment by nerzhul** [comment]
Soft fail test commit: https://travis-ci.org/minetest/minetest/builds/215263717
Hard fail test commit: https://travis-ci.org/minetest/minetest/builds/215263886

**Comment by nerzhul** [comment]
When approval will be added i will remove the two test commits and merge the first commit

**Comment by sfan5** [comment]
wouldn't it be easier to have a blacklist?

**Comment by nerzhul** [comment]
what do you want to blacklist ? the currently valid files ? it's better to dynamicly have every file checked and whitelist non formatted files, and after a file is valid, remove it from whitelist, and at the end, remove blacklist/whitelist support.
I think we should at a point fix coding style in little files/non opened/active PR files to permit remove this whitelist a little bit more faster

File: util/travis/script.sh [hunk 1]
```diff
@@ -28,15 +30,22 @@ function perform_lint() {
 			else
 				printf ":\n%s\n" "$d"
 			fi
-			# Disable build failure at this moment as we need to have a complete MT source whitelist to check
-			fail=0
+
+			whitelisted=$(egrep -c "^${f}" "${CLANG_FORMAT_WHITELIST}")
+
+			# If no failure recorded and file is not whitelisted, mark a failure
+			if [ ${fail} -eq 0 ] && [ ${whitelisted} -eq 0 ]; then
```
Comment (sfan5) [inline util/travis/script.sh]: the `${fail} -eq 0` is very confusing here as it doesn't actually matter

**Comment by sfan5** [comment]
Nevermind it's okay like this

File: util/travis/script.sh (hunk 1 above)
Comment (nerzhul) [inline util/travis/script.sh]: exact, i will remove it

**Pull request merged**